integration_test/tracing/tracing_test.go

```diff
@@ -0,0 +1,871 @@
+package tracing
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"slices"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/processor/transformer"
+
+	"go.opentelemetry.io/otel"
+
+	"github.com/rudderlabs/rudder-go-kit/stats/testhelper/tracemodel"
+	"github.com/rudderlabs/rudder-go-kit/testhelper/assert"
+	"github.com/rudderlabs/rudder-server/app"
+	"github.com/rudderlabs/rudder-server/gateway/response"
+	"github.com/rudderlabs/rudder-server/testhelper/transformertest"
+
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-server/jobsdb"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	_ "github.com/marcboeker/go-duckdb"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	kithttputil "github.com/rudderlabs/rudder-go-kit/httputil"
+	kithelper "github.com/rudderlabs/rudder-go-kit/testhelper"
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource"
+	"github.com/rudderlabs/rudder-go-kit/testhelper/rand"
+	"github.com/rudderlabs/rudder-server/runner"
+	"github.com/rudderlabs/rudder-server/testhelper/backendconfigtest"
+	"github.com/rudderlabs/rudder-server/testhelper/health"
+)
+
+func TestTracing(t *testing.T) {
+	t.Run("gateway-processor-router tracing", func(t *testing.T) {
+		config.Reset()
+		defer config.Reset()
+
+		bcServer := backendconfigtest.NewBuilder().
+			WithWorkspaceConfig(
+				backendconfigtest.NewConfigBuilder().
+					WithSource(
+						backendconfigtest.NewSourceBuilder().
+							WithID("source-1").
+							WithWriteKey("writekey-1").
+							WithConnection(
+								backendconfigtest.NewDestinationBuilder("WEBHOOK").
+									WithID("destination-1").
+									Build()).
+							Build()).
+					Build()).
+			Build()
+		defer bcServer.Close()
+
+		trServer := transformertest.NewBuilder().Build()
+		defer trServer.Close()
+
+		pool, err := dockertest.NewPool("")
+		require.NoError(t, err)
+
+		postgresContainer, err := resource.SetupPostgres(pool, t)
+		require.NoError(t, err)
+		zipkin, err := resource.SetupZipkin(pool, t)
+		require.NoError(t, err)
```

Comment:
Out of all the setup operations this is the most likely to fail so it might make sense to move it to the top.